src/spu/src/services/public/stream_fetch.rs

```diff
@@ -169,96 +169,86 @@ impl StreamFetchHandler {
         let mut leader_offset_receiver = self.leader_state.offset_listener(&self.isolation);
 
         let mut counter: i32 = 0;
-        let mut consumer_offset: Option<Offset> = if !consumer_wait {
-            // since we don't need to wait for consumer, can move consumer to same offset as last read
-            Some(last_read_offset)
-        } else {
-            None // offset for consumer
-        };
+        // since we don't need to wait for consumer, can move consumer to same offset as last read
+        let mut consumer_offset: Option<Offset> = (!consumer_wait).then(|| last_read_offset);
 
         loop {
             counter += 1;
-
             debug!(counter, ?consumer_offset, last_read_offset, "waiting");
 
             select! {
-
                 _ = self.end_event.listen() => {
                     debug!("end event has been received, terminating");
                     break;
                 },
 
-
                 changed_consumer_offset = self.consumer_offset_listener.listen() => {
+                    if changed_consumer_offset == INIT_OFFSET {
+                        continue;
+                    }
```

Comment:
In this section, I am reducing right-drift by:
- Inverting the conditional to the large `if` statement
- Using `continue` in the new `if` block to short-circuit the following code (equivalent to not entering the original `if` body).